typespec-tests/src/test/java/com/azure/resourcemanager/models/commontypes/managedidentity/ManagedIdentityManagerTests.java

```diff
@@ -0,0 +1,64 @@
+import com.azure.resourcemanager.models.commontypes.managedidentity.models.ManagedServiceIdentityType;
+import com.azure.resourcemanager.models.commontypes.managedidentity.models.UserAssignedIdentity;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.utils.ArmUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ManagedIdentityManagerTests {
+    private final ManagedIdentityManager manager = ManagedIdentityManager.authenticate(
+            ArmUtils.createTestHttpPipeline(),
+            ArmUtils.getAzureProfile());
+
+    @Test
+    public void testManagedIdentityManager() {
+        Map<String, String> tagsMap = new HashMap<>();
+        tagsMap.put("tagKey1", "tagValue1");
+        ManagedIdentityTrackedResource resource = manager.managedIdentityTrackedResources()
+                .define("identity")
+                .withRegion(Region.US_EAST)
+                .withExistingResourceGroup("test-rg")
+                .withProperties(new ManagedIdentityTrackedResourceProperties())
+                .withIdentity(new ManagedServiceIdentity().withType(ManagedServiceIdentityType.SYSTEM_ASSIGNED))
+                .withTags(tagsMap)
+                .create();
+        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED, resource.identity().type());
+        Assertions.assertNotNull(resource.identity().principalId());
+        Assertions.assertNotNull(resource.identity().tenantId());
+
+        resource = manager.managedIdentityTrackedResources().getById(resource.id());
+        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED, resource.identity().type());
+        Assertions.assertNotNull(resource.identity().principalId());
+        Assertions.assertNotNull(resource.identity().tenantId());
+
+        Map<String, UserAssignedIdentity> userAssignedIdentityMap = new HashMap<>();
+        userAssignedIdentityMap.put(
+                "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1",
+                new UserAssignedIdentity());
+        resource.update().withIdentity(new ManagedServiceIdentity()
+                        .withType(ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED)
+                        .withUserAssignedIdentities(userAssignedIdentityMap))
+                .apply();
+        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED, resource.identity().type());
+        Assertions.assertNotNull(resource.identity().principalId());
+        Assertions.assertNotNull(resource.identity().tenantId());
+        Assertions.assertNotNull(resource.identity().userAssignedIdentities());
+        Assertions.assertEquals(1, resource.identity().userAssignedIdentities().size());
+        resource.identity().userAssignedIdentities()
+                .values().forEach(item -> {
```

Comment:
Don't use forEach, you only have 1 item.

Reply:
Fixed in the new version.